### DISCUSSION TITLE
FRAGMOS - fpml mapping - Payout Cashflows

### DISCUSSION THREAD

**Discussion by JBZ-Fragmos** [finos/common-domain-model#3420] [Q&A] [answered]
## QUESTION - How to map Payout Cashflows ? ## 

What is the **fpml mapping source** for mapping **Cashflows incurred by Payout terms** ?

For instance the ones typically referred to in most swaps termsheets as **"Fixed Rate Amount", "Floating Rate Amount", "Equity Performance Amount", "Dividend Amount"** , etc.

## BACKGROUND - Failed attempts for identify proper mapping source in Rosetta ## 

As an information have checked **various possible fpml sources in Rosetta** below but **none of them seem to correspond** to the type of cashflow at target e.g. "Fixed Rate Amount", "Floating Rate Amount", "Equity Performance Amount", "Dividend Amount" , etc.

- mapping via **"TransferHistory"**
  - below are only "Fees" or "Premium" thus upfront cashflow at Trade inception, hence not the ones i'm looking for.
![image](https://github.com/finos/common-domain-model/assets/92978493/1edf2732-c82d-4222-a32a-4aac30328709)

- mapping via **"Transfer"**
  - none of the 3 types of events in mapping below would correspond to cashflows at target
![image](https://github.com/finos/common-domain-model/assets/92978493/a38280cc-0de6-46dc-a147-7d412f025452)

- mapping via **"Payout->Cashflow"**
  - which in graphical view of the model looks like best choice after all
![image](https://github.com/finos/common-domain-model/assets/92978493/e1e85ebb-7ba1-4b6c-aa3d-04be593eafbc)
  - but it looks like it is **not mapped neither ?** cf. below mapping value is _**"ignore"**_
![image](https://github.com/finos/common-domain-model/assets/92978493/d7797871-cfa0-4a67-b772-10951e8eec38)

**Comment by JBZ-Fragmos** [comment]
@mgratacos 
as you kindly proposed during the CRWG today, please provide the correct fpml source for extracting actual realized cashflows from fpml
thanks a lot in advance

**Comment by JBZ-Fragmos** [comment]
@lolabeis thank you as well for your feedback during the CRWG

**Comment by mgratacos** [comment, marked as answer]
Hi Jean-Baptiste,

As a follow-up from our conversation, the FpML message that allows you to report realized cash flows is the **tradeCashflowsAsserted** message, available in the **FpML Reporting View**. I am attaching some samples. See xxx-cashflow-assertion-xxx samples. Hope this helps. Let me know if you have any questions.
[cash-flow-matching.zip](https://github.com/user-attachments/files/15789351/cash-flow-matching.zip)


![image](https://github.com/finos/common-domain-model/assets/6411070/8ba78b01-758d-4305-bdca-d9080d175412)

**Comment by JBZ-Fragmos** [comment]
Thank you so much for your help, @Marc ***@***.***> , very appreciated !

Best regards

JB
